cartographer_ros/cartographer_ros/node.cc

```diff
@@ -666,7 +668,8 @@ void Node::RunFinalOptimization() {
   {
     for (const auto& entry : map_builder_bridge_.GetTrajectoryStates()) {
       const int trajectory_id = entry.first;
-      if (entry.second == TrajectoryState::ACTIVE) {
+      if (entry.second == TrajectoryState::ACTIVE &&
+          trajectories_scheduled_for_finish_.count(entry.first) == 0) {
```

Comment:
move the check into FinishTrajectoryUnderLock() and add error message there talking about pending finish?

Reply:
And nice catch, BTW!